File: PromizzApp.Client/src/app/shared/models/promise.model.ts
import { LookupItem, UserModel, TagItem } from '../.';

export class PromiseModel {
    public Id: number;
    public UserId: number;
    public Title: string = '';
    public Description: string = '';
    public State: LookupItem<number>;
    public Color: string = '';
    public EndDate: Date;
    public Promisee: UserModel;
    public Tags: TagItem[];

    public constructor(init?: Partial<PromiseModel>) {
        Object.assign(this, init);
    }
}

export class PromiseAddModel {
    public Title: string = '';
    public Description: string = '';
    public Promisees: Array<number>;
    public Color: string = '';
    public EndDate: Date;

    public constructor(init?: Partial<PromiseAddModel>) {
        Object.assign(this, init);
    }
}

File: Models/PromizzApp.Models/PromiseModel.cs
using System;
using System.Collections.Generic;

namespace PromizzApp.Models
{
    public class PromiseModel
    {
        public int Id { get; set; }
        
        public string Title { get; set; }

        public string Description { get; set; }

        public LookupItem<int> State { get; set; }

        public int OwnerId { get; set; }

        public string Color { get; set; }

        public DateTime EndDate { get; set; }

        public UserModel Promisee { get; set; }
    }

    public class PromiseAddModel
    {
        public string Title { get; set; }

        public string Description { get; set; }

        public string Color { get; set; }

        public List<int> Promisees { get; set; }

        public DateTime EndDate { get; set; }
    }
}
